nixos/modules/services/databases/lldap.nix

```diff
@@ -169,6 +249,198 @@ in
+                type = types.listOf types.str;
+                default = [ ];
+                description = "Groups the user would be a member of (all the groups must be specified in group config files).";
+              };
+            };
+          }
+        )
+      );
+    };
+
+    ensureGroups = mkOption {
+      description = ''
+        Create the groups defined here on service startup.
+
+        Non-default options must be added to the `ensureGroupFields` option.
+      '';
+      default = { };
+      type = types.attrsOf (
+        types.submodule (
+          { name, ... }:
+          {
+            freeformType = ensureFormat.type;
+
+            options = {
+              name = mkOption {
+                type = types.str;
+                description = "Name of the group.";
+                default = name;
+              };
+            };
+          }
+        )
+      );
+    };
+
+    ensureUserFields = mkOption {
+      description = "Extra fields for users";
+      default = { };
+      type = types.attrsOf (
+        types.submodule (
+          { name, ... }:
+          {
+            options = ensureFieldsOptions name;
+          }
+        )
+      );
+    };
+
+    ensureGroupFields = mkOption {
+      description = "Extra fields for groups";
+      default = { };
+      type = types.attrsOf (
+        types.submodule (
+          { name, ... }:
+          {
+            options = ensureFieldsOptions name;
+          }
+        )
+      );
+    };
+
+    ensureAdminUsername = mkOption {
+      type = types.str;
+      default = "admin";
+      description = ''
+        Username of the default admin user with which to connect to the LLDAP service.
+
+        By default, it is `"admin"`.
+        Extra admin users can be added using the `services.lldap.ensureUsers` option and adding them to the correct groups.
+      '';
+    };
+
+    ensureAdminPassword = mkOption {
+      type = types.nullOr types.str;
+      defaultText = "config.services.lldap.settings.ldap_user_pass";
+      default = cfg.settings.ldap_user_pass or null;
+      description = ''
+        Password of an admin user with which to connect to the LLDAP service.
+
+        By default, it is the same as the password for the default admin user 'admin'.
+        If using a password from another user, it must be managed manually.
+
+        Unsecure. Use `services.lldap.ensureAdminPasswordFile` option instead.
+      '';
+    };
+
+    ensureAdminPasswordFile = mkOption {
+      type = types.nullOr types.str;
+      defaultText = "config.services.lldap.settings.ldap_user_pass_file";
+      default = cfg.settings.ldap_user_pass_file or null;
+      description = ''
+        Path to the file containing the password of an admin user with which to connect to the LLDAP service.
+
+        By default, it is the same as the password for the default admin user 'admin'.
+        If using a password from another user, it must be managed manually.
+      '';
```

Comment:
Am I reading this right, that `ensureAdmin[Username,PasswordFile]` are the credentials, that the _bootstrap_ script itself uses to connect to lldap?